elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCancelThread.java

```diff
@@ -87,33 +85,20 @@ protected void cancelAsyncQuery() {
 
             queriesToCancel.stream()
                .forEach((tx) -> {
-                   JsonApiDocument jsonApiDoc = new JsonApiDocument();
-                   MultivaluedMap<String, String> queryParams = new MultivaluedHashMap<String, String>();
-                   RequestScope scope = new RequestScope("query", NO_VERSION, jsonApiDoc,
-                           transactionRegistry.getRunningTransaction(tx), null, queryParams,
-                           tx, elide.getElideSettings());
-                   transactionRegistry.getRunningTransaction(tx).cancel(scope);
+                       JsonApiDocument jsonApiDoc = new JsonApiDocument();
+                       MultivaluedMap<String, String> queryParams = new MultivaluedHashMap<String, String>();
+                       RequestScope scope = new RequestScope("query", NO_VERSION, jsonApiDoc,
+                               transactionRegistry.getRunningTransaction(tx), null, queryParams,
+                               tx, elide.getElideSettings());
+                       transactionRegistry.getRunningTransaction(tx).cancel(scope);
+                       List<AsyncQuery> asyncQueryList = asyncQueryCollection.stream()
+                               .filter(query -> query.getRequestId().equals(tx)).collect(Collectors.toList());
+                       asyncQueryList.forEach(asyncQuery -> asyncQueryDao.updateStatus(asyncQuery.getId(),
```

Comment:
It will be more efficient to call asyncQueryDao.updateAsyncQueryCollection which can do a bulk update